backends/cortex_m/ops/op_quantized_linear.cpp

```diff
@@ -0,0 +1,369 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include "cortex_m_ops_common.h"
+
+// Include CMSIS-NN headers with C linkage
+extern "C" {
+#include "arm_nnfunctions.h"
+}
+
+namespace cortex_m {
+namespace native {
+using KernelRuntimeContext = torch::executor::KernelRuntimeContext;
+
+class CMSISScratchBufferContext final {
+  // During AOT phase, quantized_linear_fusion_pass allocates total buffer
+  // and passes in as 'Tensor'. (Total buffer = 32-byte header + x bytes)
+  // ┌─────────────────┬─────────────────────────────────────┐
+  // │   Fat Header    │        CMSIS Workspace              │
+  // │    (32 bytes)   │         (x bytes)                   │
+  // └─────────────────┴─────────────────────────────────────┘
+  //          │                           │
+  //          │                           └─> Passed to CMSIS API
+  //          │
+  //          └─> Used by this C++ custom kernel only
+
+  // C++ Runtime:
+  // ┌─────────────────┬─────────────────────────────────────┐
+  // │   Fat Header    │        CMSIS Workspace              │
+  // │    (32 bytes)   │         (X bytes)                   │
+  // └─────────────────┴─────────────────────────────────────┘
+  // ^                 ^
+  // │                 │
+  // scratch_ptr       cmsis_workspace_ptr
+  // │                 │
+  // ▼                 ▼
+  //             arm_vector_sum_s8() writes kernel sums (with bias if avail):
+  //             [sum₀+bias₀][sum₁+bias₁][sum₂+bias₂]...[sum_{n-1}+bias_{n-1}]
```

Comment:
Nice subtest, sad can't do for `{n-1}` :-p
```suggestion
  //             [sum₀+bias₀][sum₁+bias₁][sum₂+bias₂]...[sum_{n-1}+bias_{n-1}]
```